umapi_client/functional.py

```diff
@@ -308,6 +308,32 @@ def delete_account(self):
         self.append(removeFromDomain={})
         return None
 
+    def split_groups(self, index, verb, group_type, max_groups):
+        """
+        Split a long list of groups in an add/remove command
+        :param index: index of command to split
+        :param verb: add or remove
+        :param group_type: group type - product, productConfiguration or usergroup
+        :param max_groups: max size of groups (if group list is larger than max it will be split)
+        :return:
+        """
+        if index > len(self.commands):
+            raise ArgumentError(six.text_type("Index {} not found in commands list").format(index))
+        if verb not in self.commands[index]:
+            raise ArgumentError(six.text_type("'{}' not specified in command").format(verb))
+        if group_type not in self.commands[index][verb]:
+            raise ArgumentError(six.text_type("'{}' not specified in command").format(group_type))
+        groups = self.commands[index][verb][group_type]
+        if len(groups) > max_groups:
+            updated_index = False
+            while len(groups) >= 1:
+                batch, groups = groups[0:max_groups], groups[max_groups:]
+                if not updated_index:
+                    self.commands[index][verb][group_type] = batch
+                    updated_index = True
+                else:
+                    self.commands.append({verb: {group_type: batch}})
```

Comment:
see comments in the connection file.  you can't add this at the end, because it doesn't preserve order of operations.  you should add it as the next command.